src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -115,10 +115,10 @@ public static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, INonNullTy
                 // string, int (leave unannotated)
                 // int?, T? where T : struct (add annotation)
                 // int? (error type)
-                isAnnotated = typeSymbol.IsNullableType();
+                isAnnotated = typeSymbol.IsNullableType() || (treatUnconstrainedTypeParameterAsNullable && typeSymbol.IsUnconstrainedTypeParameter());
```

Comment:
>treatUnconstrainedTypeParameterAsNullable [](start = 62, length = 41)

Could you update the examples in the comment to explain why the new parameter is useful?